oio/xcute/orchestrator.py

```diff
@@ -366,77 +350,104 @@ def process_reply(self, beanstalkd_job_id, encoded_reply):
 
         yield None
 
-    def refresh_all_beanstalkd(self):
+    def refresh_beanstalkd_workers_forever(self):
         """
-            Get all the beanstalkd and their tubes
+        Refresh beanstalkd workers by looking at the score,
+        existing tubes and tube statistics.
         """
+        beanstalkd_stats = dict()
 
         while self.running:
-            all_beanstalkd = self.conscience_client.all_services('beanstalkd')
+            beanstalkd_workers = list()
 
-            all_beanstalkd_with_tubes = {}
-            for beanstalkd in all_beanstalkd:
-                beanstalkd_addr = beanstalkd['addr']
-
-                try:
-                    beanstalkd_tubes = self.get_beanstalkd_tubes(
-                        beanstalkd_addr)
-                except ConnectionError:
-                    continue
-
-                all_beanstalkd_with_tubes[beanstalkd_addr] = (
-                    beanstalkd, beanstalkd_tubes)
+            try:
+                self._find_beanstalkd_workers(beanstalkd_workers,
```

Comment:
please return `beanstalkd_workers` and `beanstalkd_stats` so it's easier to see that they've been updated

Reply:
also why pass an empty list that should be filled?